src/transforms/scaling.jl

```diff
@@ -0,0 +1,55 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    ScaleCoords(sx, sy, sz)
+
+Transform geometry or mesh by scaling each coordinate of the vertices by 
+the corresponding factor. 
+
+## Arguments
+
+- `sx`: scaling factor for the first coordinate
+- `sy`: scaling factor for the second coordinate
+- `sz`: scaling factor for the third coordinate
+
+## Examples
+
+```julia
+ScaleCoords(1, 2, 3)
+```
+"""
+struct ScaleCoords{T<:Real} <: GeometricTransform
+  factors::AbstractVector{T}
+end
+
+function ScaleCoords(sx::T1, sy::T2, sz::T3) where {T1<:Real,T2<:Real,T3<:Real}
+  if sx ≤ 0 || sy ≤ 0 || sz ≤ 0
+    error("The scaling factors must be nonnegative.")
+  end
+  ScaleCoords([sx, sy, sz])
+end
```

Comment:
Try to use Vararg here. You can write:

```julia
ScaleCoords(factors...) = ScaleCoords{eltype(factors)}(factors)
```

Also, try to move the assertion to the inner constructor, that way all constructors will have the assertion.

Reply:
Do you mean `ScaleCoords{eltype(factors)}([factors...])`?

Reply:
```julia
julia> f(x::Vector) = sum(x)
f (generic function with 1 method)

julia> f(x...) = f(collect(x))
f (generic function with 2 methods)

julia> f(1,2,3)
6
```
You can use collect instead